overlay_gl/CMakeLists.txt

```diff
@@ -6,7 +6,9 @@
 # Overlay payload for UNIX-like systems.
 
 if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "amd64")
-	option(overlay-xcompile "Build 32 bit overlay library, necessary for the overlay to work with 32 bit processes." ON)
+	if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
+		option(overlay-xcompile "Build 32 bit overlay library, necessary for the overlay to work with 32 bit processes." ON)
+	endif()
```

Comment:
Technically a user could still set the (now non-existent) option on the command-line and it would still attempt to build it.

Maybe we should instead leave the option as it is and check the system inside the actual cross-compile block and if it detects OpenBSD, return without doing anything (and maybe print that we skip building the overlay cross-compilation because this is nonsense on such a system)

Reply:
> Technically a user could still set the (now non-existent) option on the command-line and it would still attempt to build it.

Sure, i can set `pipewire=ON` on OpenBSD and it would fail -- protecting users from their own footguns seems out of scope, I'm merely interested in sane defaults.

> Maybe we should instead leave the option as it is and check the system inside the actual cross-compile block and if it detects OpenBSD, return without doing anything (and maybe print that we skip building the overlay cross-compilation because this is nonsense on such a system)

That seemed less straight forward, i.e. I deliberately disabled/aborted/etc. the build as early as possible.

Reply:
But in the current situation the build is _not_ aborted early if that option is given. The user will just get a weird error message at some point while cmake is run (at best) or when the code is actually compiling (at worst).

I have the feeling that we might want to introduce some custom functions to declare options that allows to also specify which OS these work on and what default they have on which OS :thinking: 
That should improve the auto-generated docs and also the general option-handling in our cmake code...